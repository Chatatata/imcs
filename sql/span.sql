select Close from Quote_get('IBM', '2-Nov-2013', '5-Nov-2013');
select Close from Quote_get('IBM', '2-Nov-2013');
select Close from Quote_get('IBM', from_ts:='2-Nov-2013');
select Close from Quote_get('IBM', till_ts:='5-Nov-2013');
select Close from Quote_get('IBM');

select Close from Quote_span('IBM', 1, 3);
select Close from Quote_span('IBM', 1);
select Close from Quote_span('IBM', from_pos:=1);
select Close from Quote_span('IBM', till_pos:=3);
select Close from Quote_span('IBM');

select Close from Quote_get(array['ABB','IBM'], '2-Nov-2013', '5-Nov-2013');
select Close from Quote_get(array['ABB','IBM'], '2-Nov-2013');
select Close from Quote_get(array['ABB','IBM'], from_ts:='2-Nov-2013');
select Close from Quote_get(array['ABB','IBM'], till_ts:='5-Nov-2013');
select Close from Quote_get(array['ABB','IBM']);

select Close from Quote_span(array['ABB','IBM'], 1, 3);
select Close from Quote_span(array['ABB','IBM'], 1);
select Close from Quote_span(array['ABB','IBM'], from_pos:=1);
select Close from Quote_span(array['ABB','IBM'], till_pos:=3);
select Close from Quote_span(array['ABB','IBM']);