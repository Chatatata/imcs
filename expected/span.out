select Close from Quote_get('IBM', '2-Nov-2013', '5-Nov-2013');
          close          
-------------------------
 float4:{20.2,30.2,40.2}
(1 row)

select Close from Quote_get('IBM', '2-Nov-2013');
            close             
------------------------------
 float4:{20.2,30.2,40.2,50.5}
(1 row)

select Close from Quote_get('IBM', from_ts:='2-Nov-2013');
            close             
------------------------------
 float4:{20.2,30.2,40.2,50.5}
(1 row)

select Close from Quote_get('IBM', till_ts:='5-Nov-2013');
            close             
------------------------------
 float4:{10.5,20.2,30.2,40.2}
(1 row)

select Close from Quote_get('IBM');
               close               
-----------------------------------
 float4:{10.5,20.2,30.2,40.2,50.5}
(1 row)

select Close from Quote_span('IBM', 1, 3);
          close          
-------------------------
 float4:{20.2,30.2,40.2}
(1 row)

select Close from Quote_span('IBM', 1);
            close             
------------------------------
 float4:{20.2,30.2,40.2,50.5}
(1 row)

select Close from Quote_span('IBM', from_pos:=1);
            close             
------------------------------
 float4:{20.2,30.2,40.2,50.5}
(1 row)

select Close from Quote_span('IBM', till_pos:=3);
            close             
------------------------------
 float4:{10.5,20.2,30.2,40.2}
(1 row)

select Close from Quote_span('IBM');
               close               
-----------------------------------
 float4:{10.5,20.2,30.2,40.2,50.5}
(1 row)

select Close from Quote_get(array['ABB','IBM'], '2-Nov-2013', '5-Nov-2013');
          close          
-------------------------
 float4:{60.2}
 float4:{20.2,30.2,40.2}
(2 rows)

select Close from Quote_get(array['ABB','IBM'], '2-Nov-2013');
            close             
------------------------------
 float4:{60.2,70.5}
 float4:{20.2,30.2,40.2,50.5}
(2 rows)

select Close from Quote_get(array['ABB','IBM'], from_ts:='2-Nov-2013');
            close             
------------------------------
 float4:{60.2,70.5}
 float4:{20.2,30.2,40.2,50.5}
(2 rows)

select Close from Quote_get(array['ABB','IBM'], till_ts:='5-Nov-2013');
            close             
------------------------------
 float4:{60.2}
 float4:{10.5,20.2,30.2,40.2}
(2 rows)

select Close from Quote_get(array['ABB','IBM']);
               close               
-----------------------------------
 float4:{60.2,70.5}
 float4:{10.5,20.2,30.2,40.2,50.5}
(2 rows)

select Close from Quote_span(array['ABB','IBM'], 1, 3);
          close          
-------------------------
 float4:{70.5}
 float4:{20.2,30.2,40.2}
(2 rows)

select Close from Quote_span(array['ABB','IBM'], 1);
            close             
------------------------------
 float4:{70.5}
 float4:{20.2,30.2,40.2,50.5}
(2 rows)

select Close from Quote_span(array['ABB','IBM'], from_pos:=1);
            close             
------------------------------
 float4:{70.5}
 float4:{20.2,30.2,40.2,50.5}
(2 rows)

select Close from Quote_span(array['ABB','IBM'], till_pos:=3);
            close             
------------------------------
 float4:{60.2,70.5}
 float4:{10.5,20.2,30.2,40.2}
(2 rows)

select Close from Quote_span(array['ABB','IBM']);
               close               
-----------------------------------
 float4:{60.2,70.5}
 float4:{10.5,20.2,30.2,40.2,50.5}
(2 rows)

